### PR TITLE
1 implement error functions for machine learning algorithms

### LINKED ISSUE #1: Implement Six Error Functions for machine learning algorithms
Need to Implement following Error functions. 

- [x] Mean Square Error
- [x] Mean Absolute Error
- [x] Binary Cross Entropy

### PR DESCRIPTION
**Pull request by TharakaUmayanga**
# Description
 Need to implement following error functions

- [x] Mean Squared Error (MSE)
- [x] Mean Absolute Error (MAE)
- [x] Binary Cross-Entropy Loss

## Related Issues

Closes #1 

### Testing Done
 
checked results with few online calculators

## Checklist for the reviewer

- [x]  All existing tests are passing.
- [x] All new methods are covered by Unit Tests
- [x] Documentation has been updated
- [x] Docstrings have been included

### CODE REVIEW

**Comment by TharakaUmayanga** [comment]
@rshcs  can you please review this request

**Comment by rshcs** [comment]
> @rshcs can you please review this request

Noted and will do it.

**Review by rshcs — APPROVED** [review]
LGTM
